src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        linkLastElement(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAdd(index);
+        if (index == size) {
+            linkLastElement(value);
+        } else {
+            linkBeforeElement(value, searchForNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list.isEmpty()) {
+            return;
+        }
+        Node<T> previous = last;
+        T[] elements = (T[]) list.toArray();
+        for (T element : elements) {
+            Node<T> newNode = new Node<>(previous, element, null);
+            if (previous == null) {
+                first = newNode;
+            } else {
+                previous.next = newNode;
+                previous = newNode;
```

Comment:
The `previous` variable should be updated after setting `previous.next = newNode`. Move `previous = newNode;` outside the else block to ensure it is updated in both cases.